qemu/tests/win_driver_install_by_installer.py

```diff
@@ -0,0 +1,330 @@
+                            "VirtIO Serial Driver", "VirtIO Balloon Driver",
+                            "QEMU PVPanic Device", "VirtIO Input Driver",
+                            "VirtIO FS Device", "Red Hat VirtIO SCSI controller",
+                            "Red Hat VirtIO SCSI pass-through controller"]
+        for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                         device_name_list, device_hwid_list):
+            uninstall_driver(session, driver_name, device_name, device_hwid)
+
+        error_context.context("Install 'qemu-guest-agent' package in guest.",
+                              logging.info)
+        vol_virtio_key = "VolumeName like '%virtio-win%'"
+        vol_virtio = utils_misc.get_win_disk_vol(session, vol_virtio_key)
+        qemu_ga_pkg_path = r"%s:\%s\%s" % (vol_virtio, "guest-agent", qemu_ga_pkg)
+        gagent_install_cmd = params["gagent_install_cmd"] % qemu_ga_pkg_path
+        s_inst, o_inst = session.cmd_status_output(gagent_install_cmd)
+        if s_inst != 0:
+            test.fail("qemu-guest-agent install failed,"
+                      " the detailed info:\n%s." % o_inst)
+        expected_gagent_version = session.cmd_output(gagent_pkg_info_cmd).split()[-2]
+
+        error_context.context("Try to uninstall 'qemu-guest-agent' package.",
+                              logging.info)
+        s, o = session.cmd_status_output(gagent_uninstall_cmd)
+        if s:
+            test.fail("Could not uninstall qemu-guest-agent package "
+                      "in guest', detail: '%s'" % o)
+
+        # copy autoit scripts and installer tool to the same path
+        error_context.context("Copy autoit scripts and virtio-win-guest-tools.exe "
+                              "to the same path.", logging.info)
+        dst_path = r"C:\\"
+        installer_path = r"%s:\%s" % (vol_virtio, "virtio-win-guest-tools.exe")
+        install_script_path = utils_misc.set_winutils_letter(session,
+                                                             params["install_script_path"])
+        repaire_script_path = utils_misc.set_winutils_letter(session,
+                                                             params["repaire_script_path"])
+        uninstall_script_path = utils_misc.set_winutils_letter(session,
+                                                               params["uninstall_script_path"])
+        src_files = [installer_path, install_script_path, repaire_script_path,
+                     uninstall_script_path]
+        for src_file in src_files:
+            copy_cmd = "xcopy %s %s" % (src_file, dst_path)
+            status, output = session.cmd_status_output(copy_cmd)
+            if status != 0:
+                test.error("Copy file error,"
+                           " the detailed info:\n%s." % output)
+
+        session = vm.reboot(session)
+
+        # install winfsp tool
```

Comment:
Why need to instll WinFSP, it's not a must.
drivers should be installed even without it

Reply:
oh ，my bad, have removed  it.